xcframework binary dependency does not work from SPM/CLI (but does with Xcode)

### Description

I have a framework (which is built with resiliency with SPM) that I package as a `.xcframework` and a trivial client that links against it (to start verifying that things are built as expected, ensuring I can use @unknown default:)

It works when I run the SPM package in Xcode, but not from the command line using SPM - then it can't find the `.xcframework` / module for import.

If one then opens Package.swift and runs it in Xcode (14 b5) - things work as expected.

There's a [background discussion over at Swift Forums](https://forums.swift.org/t/spm-using-path-to-binary-target-xcframework-works-in-xcode-but-not-from-command-line/59617).

I have set up a completely minimised set of two reproducing projects and have attached step-by-step instructions on how to repeat and reproduce (expects checking out both repos side-by-side in the same directory, the example uses GitHub CLI to clone repos, but I can attached .zip files if you prefer instead).

### Expected behavior

I expected the build to succeed (as it does if I open the same `Package.swift` in Xcode and builds/runs)

### Actual behavior

Build fails:

```
hassila@max ~/t/prototype-library-evolution-client (main)> swift build
Building for debugging...
/Users/hassila/test/prototype-library-evolution-client/Sources/PrototypeLibraryEvolutionClient/PrototypeLibraryEvolutionClient.swift:2:8: error: no such module 'PrototypeLibraryEvolution'
import PrototypeLibraryEvolution
       ^
/Users/hassila/test/prototype-library-evolution-client/Sources/PrototypeLibraryEvolutionClient/PrototypeLibraryEvolutionClient.swift:2:8: error: no such module 'PrototypeLibraryEvolution'
import PrototypeLibraryEvolution
       ^

hassila@max ~/t/prototype-library-evolution-client (main) [1]> 
```


### Steps to reproduce

1. gh repo clone ordo-one/prototype-library-evolution
2. gh repo clone ordo-one/prototype-library-evolution-client
3. cd prototype-library-evolution
4. swift build
5. xcodebuild -create-xcframework -library .build/arm64-apple-macosx/debug/libPrototypeLibraryEvolution.dylib -output PrototypeLibraryEvolution.xcframework
6. cd ../prototype-library-evolution-client
7. swift build 


### Swift Package Manager version/commit hash

Swift Package Manager - Swift 5.7.0

### Swift & OS version (output of `swift --version && uname -a`)

```
hassila@max ~/t/prototype-library-evolution-client (main)> swift --version
swift-driver version: 1.62.3 Apple Swift version 5.7 (swiftlang-5.7.0.123.8 clang-1400.0.29.50)
Target: arm64-apple-macosx12.0
hassila@max ~/t/prototype-library-evolution-client (main)> uname  -a
Darwin max.local 21.6.0 Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000 arm64
hassila@max ~/t/prototype-library-evolution-client (main)> 
```

Xcode that's being used as reference where it works I Xcode 14b5.

--- a/Sources/Build/BuildPlan.swift
+++ b/Sources/Build/BuildPlan.swift
@@ -2272,8 +2272,8 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
                 if case .xcframework = target.kind {
                     let libraries = try self.parseXCFramework(for: target)
                     for library in libraries {
-                        if let headersPath = library.headersPath {
-                            clangTarget.additionalFlags += ["-I", headersPath.pathString]
+                        library.headersPaths.forEach {
+                            clangTarget.additionalFlags += ["-I", $0.pathString]
                         }
                         clangTarget.libraryBinaryPaths.insert(library.libraryPath)
                     }
@@ -2309,8 +2309,8 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
                 if case .xcframework = target.kind {
                     let libraries = try self.parseXCFramework(for: target)
                     for library in libraries {
-                        if let headersPath = library.headersPath {
-                            swiftTarget.additionalFlags += ["-Xcc", "-I", "-Xcc", headersPath.pathString]
+                        library.headersPaths.forEach {
+                            swiftTarget.additionalFlags += ["-I", $0.pathString, "-Xcc", "-I", "-Xcc", $0.pathString]
                         }
                         swiftTarget.libraryBinaryPaths.insert(library.libraryPath)
                     }

--- a/Sources/SPMBuildCore/BinaryTarget+Extensions.swift
+++ b/Sources/SPMBuildCore/BinaryTarget+Extensions.swift
@@ -23,8 +23,8 @@ public struct LibraryInfo: Equatable {
     /// The path to the binary.
     public let libraryPath: AbsolutePath
 
-    /// The path to the headers directory, if one exists.
-    public let headersPath: AbsolutePath?
+    /// The paths to the headers directories.
+    public let headersPaths: [AbsolutePath]
 }
 
 
@@ -54,8 +54,8 @@ extension BinaryTarget {
         // Construct a LibraryInfo for the library.
         let libraryDir = self.artifactPath.appending(component: library.libraryIdentifier)
         let libraryFile = try AbsolutePath(validating: library.libraryPath, relativeTo: libraryDir)
-        let headersDir = try library.headersPath.map { try AbsolutePath(validating: $0, relativeTo: libraryDir) }
-        return [LibraryInfo(libraryPath: libraryFile, headersPath: headersDir)]
+        let headersDirs = try library.headersPath.map({ [try AbsolutePath(validating: $0, relativeTo: libraryDir)] }) ?? [] + [libraryDir]
+        return [LibraryInfo(libraryPath: libraryFile, headersPaths: headersDirs)]
     }
 
     public func parseArtifactArchives(for triple: Triple, fileSystem: FileSystem) throws -> [ExecutableInfo] {

--- a/Tests/BuildTests/BuildPlanTests.swift
+++ b/Tests/BuildTests/BuildPlanTests.swift
@@ -4121,6 +4121,7 @@ final class BuildPlanTests: XCTestCase {
 
         let exeCompileArguments = try result.target(for: "exe").swiftTarget().compileArguments()
         XCTAssertMatch(exeCompileArguments, [.anySequence, "-F", "\(buildPath)", .anySequence])
+        XCTAssertMatch(exeCompileArguments, [.anySequence, "-I", "\(Pkg.appending(components: "Framework.xcframework", "\(platform)-\(arch)"))", .anySequence])
 
         let exeLinkArguments = try result.buildProduct(for: "exe").linkArguments()
         XCTAssertMatch(exeLinkArguments, [.anySequence, "-F", "\(buildPath)", .anySequence])

